assets/javascripts/osApiToken.js

```diff
@@ -0,0 +1,31 @@
+let apiToken = {
+    access_token: '',
+    expires_in: 0,
+    issued_at: 0
+};
+
+let makingRequest = false;
+
+
+export const getApiToken = () => {
+    const tokenCheckBuffer = 30 * 1000;
+    const tokenExpires = parseInt(apiToken.expires_in) * 1000 + parseInt(apiToken.issued_at);
+    if(Date.now() > tokenExpires - tokenCheckBuffer && !makingRequest){
+        getFreshApiToken();
+    }
+    return apiToken.access_token;
+}
```

Comment:
getApiToken can't be async as it needs to give an immediate response within the maplibra transformRequest method
if we detect the token is close to expiring (it has less than 30 seconds of life left) we initiate getting a new one. 
the old token should be returned straight away, and once the newer token has been retrieved that one will start to be returned